apps/web/src/state/farmsV4/state/extendPools/hooks.ts

```diff
@@ -0,0 +1,56 @@
+import { useAtom } from 'jotai'
+import { useCallback, useEffect, useState } from 'react'
+import { extendPoolsAtom, ExtendPoolsQuery, extendPoolsQueryAtom } from './atom'
+import { fetchExplorerPoolsList } from './fetcher'
+
+const RESET_QUERY_KEYS = ['protocols', 'orderBy', 'chains', 'pools', 'tokens'] as Array<keyof ExtendPoolsQuery>
+
+export const useExtendPools = () => {
+  const [query, _setQuery] = useAtom(extendPoolsQueryAtom)
+  const [extendPools, updateExtendPools] = useAtom(extendPoolsAtom)
+  const [pageEnd, setPageEnd] = useState(false)
+  const [nextCursor, setNextCursor] = useState('')
+  const setQuery = useCallback(
```

Comment:
name of `setQuery` have some confused, something like 'fetchWithQuery' will be better